okhttp-logging-interceptor/src/main/java/com/squareup/okhttp/logging/HttpLoggingInterceptor.java

```diff
@@ -220,9 +231,13 @@ private static String protocol(Protocol protocol) {
     return protocol == Protocol.HTTP_1_0 ? "HTTP/1.0" : "HTTP/1.1";
   }
 
-  private static String requestPath(HttpUrl url) {
-    String path = url.encodedPath();
-    String query = url.encodedQuery();
-    return query != null ? (path + '?' + query) : path;
+  private String requestPath(HttpUrl url) {
+    if (logFullUrl) {
```

Comment:
This logic is backwards. Also you don't need an `else` when you `return` inside an `if`
```java
if (logFullUrl) {
  return url.toString();
}

// other things
}
```